fbpic/particles/deposition/threading_methods.py

```diff
@@ -427,3 +427,34 @@ def deposit_J_prange_linear(x, y, z, w,
             j_z_m1_global[tx,iz_cell+1 + shift_z, ir_cell+1 + shift_r] += J_z_m1_11
 
     return
+
+# -----------------------------------------------------------------------
+# Parallel reduction of the global arrays for threads into a single array
+# -----------------------------------------------------------------------
+
+@numba.njit( parallel=True )
+def sum_reduce_2d_array( global_array, reduced_array ):
+    """
+    Sum the array `global_array` along its first axis and 
+    add it into `reduced_array`.
+
+    Parameters:
+    -----------
+    global_array: 3darray of complexs
+       Field array whose first dimension corresponds to the 
+       reduction dimension (typically: the number of threads used
+       during the current deposition)
+
+    reduced array: 2darray of complexs
+    """
+    # Extract size of each dimension
+    Nreduce, Nz, Nr = global_array.shape
+
+    # Parallel loop over iz
+    for iz in prange( Nz ):
+        # Loop over the reduction dimension (slow dimension)
+        for it in range( Nreduce ):
+            # Loop over ir (fast dimension)
+            for ir in range( Nr ):

```

Comment:
Wouldn't it be possible to use prange for Nr as well? In that case there might be some vectorisation possible?